manager/scheduler/constraint.go

```diff
@@ -18,14 +18,14 @@ type ConstraintFilter struct {
 
 // SetTask returns true when the filter is enable for a given task.
 func (f *ConstraintFilter) SetTask(t *api.Task) bool {
-	if t.Spec.Placement != nil && len(t.Spec.Placement.Constraints) > 0 {
-		constraints, err := ParseExprs(t.Spec.Placement.Constraints)
-		if err == nil {
-			f.constraints = constraints
-			return true
-		}
+	if t.Spec.Placement == nil || len(t.Spec.Placement.Constraints) == 0 {
+		return false
 	}
-	return false
+
+	// constraints form has been checked at controlapi
+	constraints, _ := ParseExprs(t.Spec.Placement.Constraints)
```

Comment:
If there is error, what would you suggest for this placement? Just log the error, or insert an invalid Expr?